### PR TITLE
Improvements to Genie core

### PR DESCRIPTION
**Pull request by prithvikannan**
* Return markdown format from Genie
* Add max iterations to Genie polling
* Trim Genie response if too large
* Handle Genie failed status

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -6,12 +6,14 @@ authors = [
...
 readme = "README.md"
-requires-python = ">=3.8"
+requires-python = ">=3.9"
 dependencies = [
   "typing_extensions",
   "pydantic",
   "databricks-sdk>=0.34.0",
   "pandas",
+  "tiktoken>=0.8.0",
+  "tabulate",
```
Comment (prithvikannan) [inline pyproject.toml:16]: needed for dataframe `to_markdown()`

File: src/databricks_ai_bridge/genie.py [hunk 2]
```diff
@@ -75,24 +93,34 @@ def create_message(self, conversation_id, content):
 
     def poll_for_result(self, conversation_id, message_id):
         def poll_result():
-            while True:
+            iteration_count = 0
+            while True and iteration_count < MAX_ITERATIONS:
```
Comment (aravind-segu) [inline src/databricks_ai_bridge/genie.py]: Nit: No need to have `True and` here

File: src/databricks_ai_bridge/genie.py [hunk 3]
```diff
@@ -75,24 +93,34 @@ def create_message(self, conversation_id, content):
...
+                    logging.debug("Genie failed to execute the query")
+                    return None
                 else:
                     logging.debug(f"Waiting...: {resp['status']}")
                     time.sleep(5)
 
         def poll_query_results():
-            while True:
+            iteration_count = 0
+            while True and iteration_count < MAX_ITERATIONS:
```
Comment (aravind-segu) [inline src/databricks_ai_bridge/genie.py]: Similar to above here

File: src/databricks_ai_bridge/genie.py [hunk 4]
```diff
@@ -75,24 +93,34 @@ def create_message(self, conversation_id, content):
...
-                    sql = next(r for r in resp["attachments"] if "query" in r)["query"]["query"]
+                    query = next(r for r in resp["attachments"] if "query" in r)["query"]
+                    description = query.get("description", "")
+                    sql = query.get("query", "")
+                    logging.debug(f"Description: {description}")
                     logging.debug(f"SQL: {sql}")
                     return poll_query_results()
                 elif resp["status"] == "COMPLETED":
                     return next(r for r in resp["attachments"] if "text" in r)["text"]["content"]
+                elif resp["status"] == "FAILED":
```
Comment (aravind-segu) [inline src/databricks_ai_bridge/genie.py:113]: There are a few more message statuses according to the sdk. Should we also add `CANCELLED` and `QUERY_RESULT_EXPIERED` here? 

https://databricks-sdk-py.readthedocs.io/en/latest/dbdataclasses/dashboards.html#databricks.sdk.service.dashboards.MessageStatus

**Review by aravind-segu — APPROVED** [review]
LGTM, just minor nits.

**Pull request merged**